tmt/steps/report/reportportal.py

```diff
@@ -226,12 +226,15 @@ class ReportReportPortal(tmt.steps.report.ReportPlugin[ReportReportPortalData]):
     def handle_response(self, response: requests.Response) -> None:
         """ Check the endpoint response and raise an exception if needed """
 
-        if not response.ok:
+        self.debug("Response code from the endpoint", response.status_code)
+        self.debug("Message from the endpoint", response.text)
+
+        if response.status_code == 504:
+            self.debug("Ignoring RP timeout response")
```

Comment:
Please, make it a warning. IIUIC, the user runs tests, the user expects results in RP, but the results are not there, the user turns to the TF log and sees nothing out of the ordinary because the message is in `log.txt` as a mere debug note. On the other hand, a warning in tmt output reporting there was a timeout and the submission was not repeated would provide an immediate hint.

And a question: is it really safe to not repeat the request? At least once or twice, before giving up? Seems like the results are not that important after all to me when the plugin gives up on uploading them after the first hurdle.

Reply:
According to my observation, data are in RP despite the timeout. Retry is worth trying, hopefully it won't duplicate test results in RP.

Reply:
So for now we have updated the `status_forcelist` with 504.